src/test/kotlin/com/exactpro/th2/common/schema/message/impl/rabbitmq/transport/TransportUtilsTest.kt

```diff
@@ -165,20 +178,75 @@ class TransportUtilsTest {
-                                        .setSessionAlias("alias")
-                                        .setDirection(Direction.INCOMING)
-                                        .setSequence(1)
-                                        .setTimestamp(Instant.now())
-                                }.build())
+                            .addMessage(
+                                ParsedMessage.builder()
+                                    .setType(msgType)
+                                    .setBody(emptyMap())
+                                    .apply {
+                                        idBuilder()
+                                            .setSessionAlias("alias")
+                                            .setDirection(Direction.INCOMING)
+                                            .setSequence(1)
+                                            .setTimestamp(Instant.now())
+                                    }.build()
+                            )
                             .build()
                     ).build()
                 assertSame(batch, routerFilters.filter(batch))
             },
         )
     }
+
+    @Test
+    fun `proto to transport and back test`() {
+        val transport = ParsedMessage.builder().apply {
+            idBuilder().apply {
+                setSessionAlias("test-session-alias")
+                setDirection(Direction.OUTGOING)
+                setTimestamp(Instant.now())
+                setSequence(Random.nextLong())
+                setSubsequence((1..Random.nextInt(2, 5)).map { Random.nextInt() })
+            }
+            setEventId(EventId.builder().apply {
+                setId("test-id")
+                setBook(BOOK_NAME)
+                setScope("test-scope")
+                setTimestamp(Instant.now())
+            }.build())
+            setProtocol("test-protocol")
+            setType("test-type")
+            metadataBuilder().apply {
+                put("test-property", "test-property-value")
+            }
+            bodyBuilder().apply {
+                put("null", null)
+                put("simple", "test-simple")
+                put("list", listOf("simple", null))
+                put(
+                    "map", mapOf(
+                        "null" to null,
+                        "simple" to "test-sub-simple",
+                        "list" to listOf("simple", null),
+                        "map" to mapOf(
+                            "null" to null,
+                            "simple" to "test-sub-sub-simple",
+                        )
+                    )
+                )
+            }
+            setType("test-type")
+        }.build()
+
+        val proto = transport.toProto(BOOK_NAME, SESSION_GROUP)
+        assertEquals(BOOK_NAME, proto.metadata.id.bookName)
+        assertEquals(SESSION_GROUP, proto.metadata.id.connectionId.sessionGroup)
+        proto.toTransport().apply {
+            assertEquals(transport, this)
+        }
```

Comment:
Honestly, it is not the right way to test the conversion, I think... If you want to test conversion you need separate tests for `transport -> proto` and `proto -> transport` and do not join them.

Reply:
Implemented in the common-utils project